File: backend/src/main/java/com/app/mevocab/internal/result/dto/request/SubmitRequest.java
package com.app.mevocab.internal.result.dto.request;

import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class SubmitRequest {
    String resultID;
    List<Submit> submits;
    @Data
    public static class Submit {
        Integer questionID;
        String answer;
    }
}

File: frontend/lib/data/models/dto/Request/SubmitRequest.dart
class SubmitRequest {
  final String resultID;
  final List<Submit> submits;

  SubmitRequest({
    required this.resultID,
    required this.submits,
  });

  Map<String, dynamic> toJson() {
    return {
      'resultID': resultID,
      'submits': submits.map((s) => s.toJson()).toList(),
    };
  }

  @override
  String toString() {
    return 'SubmitRequest(resultID: $resultID, submits: $submits)';
  }
}

class Submit {
  final int questionID;
  final String answer;

  Submit({
    required this.questionID,
    required this.answer,
  });

  Map<String, dynamic> toJson() {
    return {
      'questionID': questionID,
      'answer': answer,
    };
  }

  factory Submit.fromJson(Map<String, dynamic> json) {
    return Submit(
      questionID: json['questionID'],
      answer: json['answer'],
    );
  }
  @override
  String toString() {
    return 'Submit(questionID: $questionID, answer: $answer)';
  }
}
